### DISCUSSION TITLE
Here's a note book for running on Google Collab

### DISCUSSION THREAD

**Discussion by bbartling** [PromtEngineer/localGPT#228] [Show and tell]
https://gist.github.com/bbartling/5172f46350af5e7a97c8363888f76f73

I had to upgrade to the Pro version as the free version of Google collab did not have enough RAM. Code seems to work let me know if there is anything funny at all with this approach not a lot of wisdom here

**Comment by huycke** [comment]
> not a lot of wisdom here

This resonates with me

**Reply by creuzerm** [reply]
Nice. I may give this a try running as a local Jupyterlab.

What was the question response time? No timestamps in that section of the notebook, I can't guess at the speed.

**Reply by bbartling** [reply]
One thing to note is for what ever reason the first time I ran that notebook it worked. There was a 100+ gigs of RAM available on Google collab Pro (my first time trying it) and the next couple times I ran it there was only about ~30 Gigs of ram which will fail the code because there isnt enought Ram for Cuda.

Also it works without the Auto GPT git clone as well, not sure why that is needed but all the code was captured from this repo.

Am curious to tinker with this on Torent GPT, maybe ill post an update here if I can get this collab notebook to work with Torent GPT

**Reply by huycke** [reply]
I've been frustrated trying to work with colab pro. Maybe pro+ users don't experience this, but sometimes, like you note, there isn't enough ram and I cannot ever get something that needs to run over a few hours to do so without at some point timing out, crashing, or getting dc'd.
